frontend/src/lib/StaticGraphParser.ts

```diff
@@ -52,214 +48,105 @@ export class SelectedNodeInfo {
-    info.volumeMounts = (template.container.volumeMounts || []).map(v => [v.mountPath, v.name]);
-  } else {
-    info.nodeType = 'resource';
-    if (template.resource && template.resource.action && template.resource.manifest) {
-      info.resource = [[template.resource.action, template.resource.manifest]];
-    } else {
-      info.resource = [[]];
-    }
+  info.nodeType = 'container';
+  if (task['taskSpec'] && task['taskSpec']['steps']){
+    const steps = task['taskSpec']['steps']
+    info.args = steps[0]['args'] || [];
+    info.command = steps[0]['command'] || [];
+    info.image = steps[0]['image'] || [];
+    info.volumeMounts = (steps[0]['volumeMounts'] || []).map((volume: any) => [volume.mountPath, volume.name])
   }
 
-  if (template.inputs) {
-    info.inputs = (template.inputs.parameters || []).map(p => [p.name, p.value || '']);
-  }
-  if (template.outputs) {
-    info.outputs = (template.outputs.parameters || []).map(p => {
-      let value = '';
-      if (p.value) {
-        value = p.value;
-      } else if (p.valueFrom) {
-        value =
-          p.valueFrom.jqFilter ||
-          p.valueFrom.jsonPath ||
-          p.valueFrom.parameter ||
-          p.valueFrom.path ||
-          '';
-      }
-      return [p.name, value];
-    });
-  }
+  if (task['taskSpec'] && task['taskSpec']['params'])
+    info.inputs = (task['taskSpec']['params'] || []).map((p: any) => [p['name'], p['value'] || '']);
+  if (task['taskSpec']['results'])
+    info.outputs = (task['taskSpec']['results'] || []).map((p: any) => {return [p['name'], p['description'] || '']});
 
   return info;
 }
 
-/**
- * Recursively construct the static graph of the Pipeline.
- *
- * @param graph The dagre graph object
- * @param rootTemplateId The current template being explored at this level of recursion
- * @param templates An unchanging map of template name to template and type
- * @param parentFullPath The task that was being examined when the function recursed. This string
- * includes all of the nodes above it. For example, with a graph such as:
- *     A
- *    / \
- *   B   C
- *      / \
- *     D   E
- * where A and C are DAGs, the parentFullPath when rootTemplateId is C would be: /A/C
- */
-function buildDag(
-  graph: dagre.graphlib.Graph,
-  rootTemplateId: string,
-  templates: Map<string, { nodeType: nodeType; template: Template }>,
-  alreadyVisited: Map<string, string>,
-  parentFullPath: string,
-): void {
-  const root = templates.get(rootTemplateId);
+export function createGraph(workflow: any): dagre.graphlib.Graph {
+  const graph = new dagre.graphlib.Graph();
+  graph.setGraph({});
+  graph.setDefaultEdgeLabel(() => ({}));
 
-  if (root && root.nodeType === 'dag') {
-    // Mark that we have visited this DAG, and save the original qualified path to it.
-    alreadyVisited.set(rootTemplateId, parentFullPath || '/' + rootTemplateId);
-    const template = root.template;
+  buildTektonDag(graph, workflow);
```

Comment:
come back to see if there can be a generic buildDag() to then call specific implementations based on Tekton or Argo